ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Node/Serialisation.hs

```diff
@@ -292,20 +295,18 @@ instance ShelleyBasedEra era => SerialiseNodeToClient (ShelleyBlock proto era) (
   encodeNodeToClient _ _ = toEraCBOR @era
   decodeNodeToClient _ _ = fromEraCBOR @era
 
-instance ShelleyCompatible proto era
-      => SerialiseNodeToClient (ShelleyBlock proto era) (SomeSecond BlockQuery (ShelleyBlock proto era)) where
-  encodeNodeToClient _ version (SomeSecond q)
+instance (ShelleyCompatible proto era, LedgerSupportsProtocol (ShelleyBlock proto era))
+      => SerialiseNodeToClient (ShelleyBlock proto era) (SomeBlockQuery (BlockQuery (ShelleyBlock proto era))) where
+  encodeNodeToClient _ version (SomeBlockQuery q)
     | blockQueryIsSupportedOnVersion q version
     = encodeShelleyQuery q
     | otherwise
-    = throw $ ShelleyEncoderUnsupportedQuery (SomeSecond q) version
+    = throw $ ShelleyEncoderUnsupportedQuery (SomeBlockQuery q) version
```

Comment:
This will change the encoding of unsupported queries in this version. Is that important?